src/resolvers/resolvers.ts

```diff
@@ -0,0 +1,23 @@
+import { GraphQLDate, GraphQLDateTime } from 'graphql-iso-date';
+import { connection } from '../database/connection';
+import { NotablePerson } from '../database/entities/notablePerson';
+
+export const resolvers = {
+  DateTime: GraphQLDateTime,
+
+  DateOnly: GraphQLDate,
+
+  RootQuery: {
+    async notablePerson(_: any, { slug }: { slug: string }) {
```

Comment:
Why is the first `_` argument needed?

Reply:
This is return value of the parent query, for the root query it's `undefined`.

Sometimes you want granular control over each field in the query, for example, in the following query:

```graphql
query {
  notablePerson(slug: "Tom_Hanks") {
    slug
    name
    events {
      id
      text
      sourceUrl
      isQuoteByNotablePerson
      addedAt
      happenedOn
    }
  }
}
```

You can split the logic for resolving `notablePerson.events` into a separate function, but you still need to know the ID of the notable person to find the events for. The ID can be read from the parent result, which would the first argument. It would look like this:

```graphql
RootQuery: {
 notablePerson: {
    resolve(): // Return { id, slug... }
    fields: {
       events(parent) {
          // Find events for parent.id
       }
    }
 }
}
```